chart/src/adapters.rs

```diff
@@ -0,0 +1,224 @@
+        self.0.write(key).remove(key).is_some()
+    }
+
+    fn update(&mut self, key: &Self::Key) -> bool {
+        let mut map = self.0.write(key);
+        map.get_mut(key).map(|v| *v += 1).is_some()
+    }
+}
+
+#[derive(Clone)]
+pub struct CHashMapTable<K>(Arc<CHashMap<K, u32>>);
+
+impl<K> Collection for CHashMapTable<K>
+where
+    K: Send + Sync + From<u64> + Copy + 'static + std::hash::Hash + Eq + std::fmt::Debug,
+{
+    type Handle = Self;
+    fn with_capacity(capacity: usize) -> Self {
+        Self(Arc::new(CHashMap::with_capacity(capacity)))
+    }
+
+    fn pin(&self) -> Self::Handle {
+        self.clone()
+    }
+}
+
+impl<K> CollectionHandle for CHashMapTable<K>
+where
+    K: Send + Sync + From<u64> + Copy + 'static + std::hash::Hash + Eq + std::fmt::Debug,
+{
+    type Key = K;
+
+    fn get(&mut self, key: &Self::Key) -> bool {
+        self.0.get(key).is_some()
+    }
+
+    fn insert(&mut self, key: &Self::Key) -> bool {
+        self.0.insert(*key, 0).is_none()
+    }
+
+    fn remove(&mut self, key: &Self::Key) -> bool {
+        self.0.remove(key).is_some()
+    }
+
+    fn update(&mut self, key: &Self::Key) -> bool {
+        self.0
+            .get_mut(key)
+            .map(|mut r| {
+                *r += 1;
+            })
+            .is_some()
+    }
+}
+
+#[derive(Clone)]
+pub struct ContrieTable<K: Eq + std::hash::Hash + 'static>(Arc<ConMap<K, Mutex<u32>>>);
+
+impl<K> Collection for ContrieTable<K>
+where
+    K: Send + Sync + From<u64> + Copy + 'static + std::hash::Hash + Eq + std::fmt::Debug,
+{
+    type Handle = Self;
+    fn with_capacity(_: usize) -> Self {
+        Self(Arc::new(ConMap::new()))
+    }
+
+    fn pin(&self) -> Self::Handle {
+        self.clone()
+    }
+}
+
+impl<K> CollectionHandle for ContrieTable<K>
+where
+    K: Send + Sync + From<u64> + Copy + 'static + std::hash::Hash + Eq + std::fmt::Debug,
+{
+    type Key = K;
+
+    fn get(&mut self, key: &Self::Key) -> bool {
+        self.0.get(key).is_some()
+    }
+
+    fn insert(&mut self, key: &Self::Key) -> bool {
+        self.0.insert(*key, Mutex::new(0)).is_none()
+    }
+
+    fn remove(&mut self, key: &Self::Key) -> bool {
+        self.0.remove(key).is_some()
+    }
+
+    fn update(&mut self, key: &Self::Key) -> bool {
+        self.0
+            .get(key)
+            .map(|e| {
+                *e.value().lock().unwrap() += 1;
+            })
+            .is_some()
+    }
+}
+
+#[derive(Clone)]
+pub struct FlurryTable(Arc<flurry::HashMap<u64, u32, FxBuildHasher>>);
+
+pub struct FlurryHandle(flurry::HashMapRef<'static, u64, u32, FxBuildHasher>);
+
+impl Collection for FlurryTable {
+    type Handle = FlurryHandle;
+
+    fn with_capacity(capacity: usize) -> Self {
+        Self(Arc::new(flurry::HashMap::with_capacity_and_hasher(
+            capacity,
+            FxBuildHasher::default(),
+        )))
+    }
+
+    fn pin(&self) -> Self::Handle {
+        unsafe { std::mem::transmute(self.0.pin()) }
```

Comment:
Yeah, this will end up never releasing garbage. You'd have to explicitly repin it to release garbage, which there isn't really a good way to do in flurry today.